spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,429 @@
+          .defaultValue("spark-kubernetes-operator")
+          .enableDynamicOverride(false)
+          .build();
+
+  public static final ConfigOption<String> OPERATOR_NAMESPACE =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.namespace")
+          .typeParameterClass(String.class)
+          .description("Namespace that operator is deployed within.")
+          .defaultValue("default")
+          .enableDynamicOverride(false)
+          .build();
+
+  public static final ConfigOption<String> OPERATOR_WATCHED_NAMESPACES =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.watchedNamespaces")
+          .description(
+              "Comma-separated list of namespaces that the operator would be "
+                  + "watching for Spark resources. If unset, operator would "
+                  + "watch all namespaces by default.")
+          .defaultValue(null)
+          .typeParameterClass(String.class)
+          .build();
+
+  public static final ConfigOption<Boolean> TERMINATE_ON_INFORMER_FAILURE =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.terminateOnInformerFailure")
+          .typeParameterClass(Boolean.class)
+          .description(
+              "Enable to indicate informer errors should stop operator startup. If "
+                  + "disabled, operator startup will ignore recoverable errors, "
+                  + "caused for example by RBAC issues and will retry "
+                  + "periodically.")
+          .defaultValue(false)
+          .enableDynamicOverride(false)
+          .build();
+
+  public static final ConfigOption<Integer> RECONCILER_TERMINATION_TIMEOUT_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.reconciler.terminationTimeoutSeconds")
+          .description(
+              "Grace period for operator shutdown before reconciliation threads are killed.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Integer.class)
+          .defaultValue(30)
+          .build();
+
+  public static final ConfigOption<Integer> RECONCILER_PARALLELISM =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.reconciler.parallelism")
+          .description(
+              "Thread pool size for Spark Operator reconcilers. Use -1 for unbounded pool.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Integer.class)
+          .defaultValue(30)
+          .build();
+
+  public static final ConfigOption<Long> RECONCILER_FOREGROUND_REQUEST_TIMEOUT_SECONDS =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.reconciler.foregroundRequestTimeoutSeconds")
+          .description(
+              "Timeout (in seconds) to for requests made to API server. this "
+                  + "applies only to foreground requests.")
+          .defaultValue(120L)
```

Comment:
Do we have any background for this 2 minutes?

Reply:
This is a practice that's yield from prod, which might vary among cluster / load. Some cloud providers (like eks) may already timeout request at 30s by default.  We can also further reduce default value this as well.